src/core/meshcop/commissioner.cpp

```diff
@@ -1169,6 +1169,18 @@ const char *Commissioner::StateToString(otCommissionerState aState)
     return str;
 }
 
+void Commissioner::LogJoinerEntry(const char *aAction, const Joiner &aJoiner) const
+{
+    otLogInfoMeshCoP("%s Joiner (%s, %s)", aAction, aJoiner.mAny ? "*" : aJoiner.mEui64.ToString().AsCString(),
+                     aJoiner.mPsk);
+}
+
+#else
+
+void Commissioner::LogJoinerEntry(const char *, const Joiner &) const
+{
+}
+
 #endif // (OPENTHREAD_CONFIG_LOG_LEVEL >= OT_LOG_LEVEL_INFO) && (OPENTHREAD_CONFIG_LOG_MLE == 1)
```

Comment:
Not related to this PR, but should this conditional be on `OPENTHREAD_CONFIG_LOG_MESHCOP`?

Reply:
Indeed. It does not seem to be correct. Will fix it. Thanks.

Reply:
Turns out we don't actually have `OPENTHREAD_CONFIG_LOG_MESHCOP` and the  meshcop logging macros are tied to the `OPENTHREAD_CONFIG_LOG_MLE` config:
https://github.com/openthread/openthread/blob/3b362071e594375035e063c0c25988b1aa52bd5f/src/core/common/logging.hpp#L291-L303

I think we may want to address this, but probably from a different PR.

Reply:
Good catch. We should clean this up in a separate PR.